### ISSUE TITLE
freqtrade.commands.trade_commands - ERROR - module 'numpy' has no attribute 'float'

### ISSUE THREAD

**Issue by tomjrtsmith** [freqtrade/freqtrade#8023]
## Describe your environment

  * Operating system: ____
  * Python Version: _____ (`python -V`) 3.10
  * CCXT version: _____ (`pip freeze | grep ccxt`)
  * Freqtrade Version: ____ (`freqtrade -V` or `docker compose run --rm freqtrade -V` for Freqtrade running in docker) Just updated for Binance Futures
  
Note: All issues other than enhancement requests will be closed without further comment if the above template is deleted or not filled out.

## Describe the problem:

*Explain the problem you have encountered*

### Steps to reproduce:

  1. _____ freqtrade.commands.trade_commands - ERROR - module 'numpy' has no attribute 'float'
  2. _____
  3. _____
  
### Observed Results:

This happened on a new install the other day as well and I assumed it was a Tensorflow issue because I've also had Tensorflow issues on that VPS but this looks like its Numpy alone causing it. This was after upgrading for Binance

  * What happened?
  * What did you expect to happen?

### Relevant code exceptions or logs

Note: Please copy/paste text of the messages, no screenshots of logs please.

  ```
  // paste your log here
  ```

**Comment by xmatthias** [comment]
we'll need full logs please (as it happens at startup, best all logs until the error happens).

just one line won't tell us where the error came from.

**Comment by tomjrtsmith** [comment]
Jan 16 05:33:28 instance-20221009-1127 systemd[1]: Starting Freqtrade Bot
FT1A...
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]: 2023-01-16
05:33:31,068 - freqtrade - INFO - freqtrade 2023.1.dev
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]: 2023-01-16
05:33:31,549 - freqtrade.worker - INFO - Starting worker 2023.1.dev
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]: 2023-01-16
05:33:31,550 - freqtrade.configuration.load_config - INFO - Using config:
/home/ubuntu/freqtrade/config.ft1a.json ...
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]: 2023-01-16
05:33:31,551 - freqtrade.loggers - INFO - Verbosity set to 0
Jan 16 05:33:31 instance-20221009-1127 freqtrade.loggers - INFO - Verbosity
set to 0
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]: 2023-01-16
05:33:31,552 - freqtrade.configuration.configuration - INFO - Runmode set
to dry_run.
Jan 16 05:33:31 instance-20221009-1127
freqtrade.configuration.configuration - INFO - Runmode set to dry_run.
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]: 2023-01-16
05:33:31,552 - freqtrade.configuration.configuration - INFO - Dry run is
enabled
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]: 2023-01-16
05:33:31,552 - freqtrade.configuration.configuration - INFO - Using DB:
"sqlite:////home/ubuntu/freqtrade/tradesv3.dwt_longshort.sqlite"
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]: 2023-01-16
05:33:31,552 - freqtrade.configuration.configuration - INFO - Using
max_open_trades: 20 ...
Jan 16 05:33:31 instance-20221009-1127
freqtrade.configuration.configuration - INFO - Dry run is enabled
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]: 2023-01-16
05:33:31,552 - freqtrade.configuration.configuration - INFO - Using
user-data directory: /home/ubuntu/freqtrade/user_data ...
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]: 2023-01-16
05:33:31,553 - freqtrade.configuration.configuration - INFO - Using data
directory: /home/ubuntu/freqtrade/user_data/data/binance ...
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]: 2023-01-16
05:33:31,553 - freqtrade.exchange.check_exchange - INFO - Checking
exchange...
Jan 16 05:33:31 instance-20221009-1127
freqtrade.configuration.configuration - INFO - Using DB:
"sqlite:////home/ubuntu/freqtrade/tradesv3.dwt_longshort.sqlite"
Jan 16 05:33:31 instance-20221009-1127
freqtrade.configuration.configuration - INFO - Using max_open_trades: 20 ...
Jan 16 05:33:31 instance-20221009-1127
freqtrade.configuration.configuration - INFO - Using user-data directory:
/home/ubuntu/freqtrade/user_data ...
Jan 16 05:33:31 instance-20221009-1127
freqtrade.configuration.configuration - INFO - Using data directory:
/home/ubuntu/freqtrade/user_data/data/binance ...
Jan 16 05:33:31 instance-20221009-1127 freqtrade.exchange.check_exchange -
INFO - Checking exchange...
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]: 2023-01-16
05:33:31,568 - freqtrade.exchange.check_exchange - INFO - Exchange
"binance" is officially supported by the Freqtrade development team.
Jan 16 05:33:31 instance-20221009-1127 freqtrade.exchange.check_exchange -
INFO - Exchange "binance" is officially supported by the Freqtrade
development team.
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]: 2023-01-16
05:33:31,568 - freqtrade.configuration.configuration - INFO - Using
pairlist from configuration.
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]: 2023-01-16
05:33:31,568 - freqtrade.configuration.deprecated_settings - WARNING -
DEPRECATED: The `forcebuy_enable` setting is deprecated and will be removed
in the next versions of Freqtrade. Please use the `force_entry_enable`
setting in your configuration instead.
Jan 16 05:33:31 instance-20221009-1127
freqtrade.configuration.configuration - INFO - Using pairlist from
configuration.
Jan 16 05:33:31 instance-20221009-1127
freqtrade.configuration.deprecated_settings - WARNING - DEPRECATED: The
`forcebuy_enable` setting is deprecated and will be removed in the next
versions of Freqtrade. Please use the `force_entry_enable` setting in your
configuration instead.
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]: 2023-01-16
05:33:31,875 - freqtrade.commands.trade_commands - ERROR - module 'numpy'
has no attribute 'float'
Jan 16 05:33:31 instance-20221009-1127 freqtrade.commands.trade_commands -
ERROR - module 'numpy' has no attribute 'float'
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]: 2023-01-16
05:33:31,876 - freqtrade.commands.trade_commands - ERROR - Fatal exception!
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]: Traceback (most
recent call last):
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:   File
"/home/ubuntu/freqtrade/freqtrade/commands/trade_commands.py", line 18, in
start_trading
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:     worker =
Worker(args)
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:   File
"/home/ubuntu/freqtrade/freqtrade/worker.py", line 37, in __init__
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:
self._init(False)
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:   File
"/home/ubuntu/freqtrade/freqtrade/worker.py", line 53, in _init
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:
self.freqtrade = FreqtradeBot(self._config)
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:   File
"/home/ubuntu/freqtrade/freqtrade/freqtradebot.py", line 63, in __init__
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:
self.strategy: IStrategy = StrategyResolver.load_strategy(self.config)
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:   File
"/home/ubuntu/freqtrade/freqtrade/resolvers/strategy_resolver.py", line 48,
in load_strategy
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:     strategy:
IStrategy = StrategyResolver._load_strategy(
Jan 16 05:33:31 instance--20221009-1127 freqtrade_a[748349]:   File
"/home/ubuntu/freqtrade/freqtrade/resolvers/strategy_resolver.py", line
256, in _load_strategy
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:     strategy =
StrategyResolver._load_object(
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:   File
"/home/ubuntu/freqtrade/freqtrade/resolvers/iresolver.py", line 148, in
_load_object
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:     (module,
module_path) = cls._search_object(directory=_path,
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:   File
"/home/ubuntu/freqtrade/freqtrade/resolvers/iresolver.py", line 130, in
_search_object
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:     obj =
next(cls._get_valid_object(module_path, object_name), None)
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:   File
"/home/ubuntu/freqtrade/freqtrade/resolvers/iresolver.py", line 91, in
_get_valid_object
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:
spec.loader.exec_module(module)  # type: ignore # importlib does not use
typehints
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:   File "<frozen
importlib._bootstrap_external>", line 883, in exec_module
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:   File "<frozen
importlib._bootstrap>", line 241, in _call_with_frames_removed
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:   File
"/home/ubuntu/freqtrade/user_data/strategies/strategies/kucoin/DWT_lev_short.py",
line 51, in <module>
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:     class
DWT_lev_short(IStrategy):
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:   File
"/home/ubuntu/freqtrade/user_data/strategies/strategies/kucoin/DWT_lev_short.py",
line 233, in DWT_lev_short
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:     def
model(self, a: np.ndarray) -> np.float:
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:   File
"/home/ubuntu/freqtrade/.env/lib/python3.10/site-packages/numpy/__init__.py",
line 284, in __getattr__
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]:     raise
AttributeError("module {!r} has no attribute "
Jan 16 05:33:31 instance-20221009-1127 freqtrade_a[748349]: AttributeError:
module 'numpy' has no attribute 'float'
Jan 16 05:33:31 instance-20221009-1127 freqtrade.commands.trade_commands -
ERROR - Fatal exception!#012Traceback (most recent call last):#012  File
"/home/ubuntu/freqtrade/freqtrade/commands/trade_commands.py", line 18, in
start_trading#012    worker = Worker(args)#012  File
"/home/ubuntu/freqtrade/freqtrade/worker.py", line 37, in __init__#012
 self._init(False)#012  File "/home/ubuntu/freqtrade/freqtrade/worker.py",
line 53, in _init#012    self.freqtrade = FreqtradeBot(self._config)#012
 File "/home/ubuntu/freqtrade/freqtrade/freqtradebot.py", line 63, in
__init__#012    self.strategy: IStrategy =
StrategyResolver.load_strategy(self.config)#012  File
"/home/ubuntu/freqtrade/freqtrade/resolvers/strategy_resolver.py", line 48,
in load_strategy#012    strategy: IStrategy =
StrategyResolver._load_strategy(#012  File
"/home/ubuntu/freqtrade/freqtrade/resolvers/strategy_resolver.py", line
256, in _load_strategy#012    strategy = StrategyResolver._load_object(#012
 File "/home/ubuntu/freqtrade/freqtrade/resolvers/iresolver.py", line 148,
in _load_object#012    (module, module_path) =
cls._search_object(directory=_path,#012  File
"/home/ubuntu/freqtrade/freqtrade/resolvers/iresolver.py", line 130, in
_search_object#012    obj = next(cls._get_valid_object(module_path,
object_name), None)#012  File
"/home/ubuntu/freqtrade/freqtrade/resolvers/iresolver.py", line 91, in
_get_valid_object#012    spec.loader.exec_module(module)  # type: ignore #
importlib does not use typehints#012  File "<frozen
importlib._bootstrap_external>", line 883, in exec_module#012  File
"<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed#012
 File
"/home/ubuntu/freqtrade/user_data/strategies/strategies/kucoin/DWT_lev_short.py",
line 51, in <module>#012    class DWT_lev_short(IStrategy):#012  File
"/home/ubuntu/freqtrade/user_data/strategies/strategies/kucoin/DWT_lev_short.py",
line 233, in DWT_lev_short#012    def model(self, a: np.ndarray) ->
np.float:#012  File
"/home/ubuntu/freqtrade/.env/lib/python3.10/site-packages/numpy/__init__.py",
line 284, in __getattr__#012    raise AttributeError("module {!r} has no
attribute "#012AttributeError: module 'numpy' has no attribute 'float'
Jan 16 05:33:32 instance-20221009-1127 systemd[1]: ft1a.service: Failed
with result 'protocol'.
Jan 16 05:33:32 instance-20221009-1127 systemd[1]: Failed to start
Freqtrade Bot FT1A.
Jan 16 05:33:32 instance-20221009-1127 systemd[1]: ft1a.service: Consumed
4.200s CPU time.
Jan 16 05:33:42 instance-20221009-1127 systemd[1]: ft1a.service: Scheduled
restart job, restart counter is at 1.
Jan 16 05:33:42 instance-20221009-1127 systemd[1]: Stopped Freqtrade Bot
FT1A.
Jan 16 05:33:42 instance-20221009-1127 systemd[1]: ft1a.service: Consumed
4.200s CPU time.
Jan 16 05:33:42 instance-20221009-1127 systemd[1]: Starting Freqtrade Bot

**Comment by tomjrtsmith** [comment]
I don't know why it would be looking for an object in DWT_lev_short.py as the strat its using is DWT_LongShort so its going and looking for objects in files in the same folder that aren't called from the strat

**Comment by xmatthias** [comment]
The strategy resolver needs to look at every file until it finds the matching class.

`DWT_lev_short.py` is before `DWT_LongShort.py` alphabetically, so the loading of this makes perfect sense.

now it shouldn't impede the progression of loading the appropriate strategy (but only fail if the strategy you're looking for has the error) - but a quick fix is clearly to not use `np.float` (which used to be an alias for the builtin float in prior numpy versions).


it's however not a recent update that's causing this - numpy deprecated this years ago - and removed this setting months ago - which suggests that your dependencies weren't matching prior to this update.

**Comment by tomjrtsmith** [comment]
The strat isn't using np.float from what I can see, nor is the other one that failed. I found this, https://stackoverflow.com/questions/74844262/how-to-solve-error-numpy-has-no-attribute-float-in-python but it didn't solve any problems either as I couldn't find any of the aliases in the strats. I've reinstalled pywavelets and scipy. 

Another question, if I'm using the precision filter is upgrading absolutely necessary?
